clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2405,139 +2307,92 @@ class SyclKernelArgsSizeChecker : public SyclKernelFieldHandler {
     addParam(FieldTy);
     return true;
   }
-
-  bool handleSyclStreamType(FieldDecl *FD, QualType FieldTy) final {
-    addParam(FieldTy);
-    return true;
-  }
-  bool handleSyclStreamType(const CXXRecordDecl *, const CXXBaseSpecifier &,
-                            QualType FieldTy) final {
-    addParam(FieldTy);
-    return true;
-  }
   using SyclKernelFieldHandler::handleSyclHalfType;
 };
 
-enum class KernelArgDescription {
-  BaseClass,
-  DecomposedMember,
-  WrappedPointer,
-  WrappedArray,
-  Accessor,
-  AccessorBase,
-  Sampler,
-  Stream,
-  KernelHandler,
-  None
-};
-
-StringRef getKernelArgDesc(KernelArgDescription Desc) {
-  switch (Desc) {
-  case KernelArgDescription::BaseClass:
-    return "Compiler generated argument for base class,";
-  case KernelArgDescription::DecomposedMember:
-    return "Compiler generated argument for decomposed struct/class,";
-  case KernelArgDescription::WrappedPointer:
-    return "Compiler generated argument for nested pointer,";
-  case KernelArgDescription::WrappedArray:
-    return "Compiler generated argument for array,";
-  case KernelArgDescription::Accessor:
-    return "Compiler generated argument for accessor,";
-  case KernelArgDescription::AccessorBase:
-    return "Compiler generated argument for accessor base class,";
-  case KernelArgDescription::Sampler:
-    return "Compiler generated argument for sampler,";
-  case KernelArgDescription::Stream:
-    return "Compiler generated argument for stream,";
-  case KernelArgDescription::KernelHandler:
-    return "Compiler generated argument for SYCL2020 specialization constant";
-  case KernelArgDescription::None:
+std::string getKernelArgDesc(std::string KernelArgDescription) {
+  if (KernelArgDescription == ":")
     return "";
-  }
-  llvm_unreachable(
-      "switch should cover all possible values for KernelArgDescription");
+  return ("Compiler generated argument for " + KernelArgDescription + ",");
 }
 
 class SyclOptReportCreator : public SyclKernelFieldHandler {
   SyclKernelDeclCreator &DC;
   SourceLocation KernelInvocationLoc;
 
   void addParam(const FieldDecl *KernelArg, QualType KernelArgType,
-                KernelArgDescription KernelArgDesc) {
+                std::string KernelArgDescription) {
```

Comment:
There's an odd mixture of styles for this kernel argument description. In some cases, it's a type name, and in other cases, it's like `DecomposedMember` which is an implementation detail more than something we'd want to expose to users. I think we should make sure that the descriptions passed here are all ones that look reasonable showing up in the opt reports. I'll comment below on some test cases.

Reply:
Are the changes in the tests enough? Would be nice if we could just used the type but we already had this discussion :)

Reply:
Yup, the changes now look good now, thanks!